### ISSUE TITLE
Dockerfile build fails

### ISSUE THREAD

**Issue by matpen** [containers/fuse-overlayfs#107]
Building the library using `builda bud` as explained in the docs 
https://github.com/containers/fuse-overlayfs/blob/f8ba9ade20d107d3e6a3dc11b6e01116894d2387/README.md#L36

fails at [this line](https://github.com/containers/fuse-overlayfs/blob/4dc60f0e157293f8692446e35d3bfc7420e36773/Dockerfile.static#L12) with error:
```
[...]
Installing /build/libfuse/doc/mount.fuse3.8 to /usr/share/man/man8
Installing /build/libfuse/build/meson-private/fuse3.pc to /usr/lib64/pkgconfig
Running custom install script '/build/libfuse/util/install_helper.sh /etc /usr/bin /usr/lib/udev/rules.d true'
mknod: /dev/fuse: Operation not permitted
FAILED: meson-install
/usr/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.
```

As far as I understand the container runtime (rightfully) prevents [this call](https://github.com/libfuse/libfuse/blob/f0e08cc700d629da2d46def8b620b0ed858cc0d9/util/install_helper.sh#L35) to `mknod`. If run in privileged mode, the call would probably succeed, but that is probably unsafe and will likely modify the host; on the other hand, this might be desired (in my case, the host already has FUSE2 installed, so it makes no difference).

Instead, I simply "tricked" the script into [believing](https://github.com/libfuse/libfuse/blob/f0e08cc700d629da2d46def8b620b0ed858cc0d9/util/install_helper.sh#L33) that the device already exists by adding the line `touch /dev/fuse` before [this one](https://github.com/containers/fuse-overlayfs/blob/4dc60f0e157293f8692446e35d3bfc7420e36773/Dockerfile.static#L12). With this workaround the build completes successfully.

Further problems I encountered:
* [this command](https://github.com/containers/fuse-overlayfs/blob/4dc60f0e157293f8692446e35d3bfc7420e36773/Dockerfile.static#L21) does not seem to work: `fuse-overlayfs: upperdir not specified`; as far as I understand, this is expected.
* it might be helpful to newcomers to explain what to do after the build (grab the binary from the image and install it somewhere in the PATH on the host).

**Comment by rhatdan** [comment]
@matpen could you try with buildah in master with the --device /dev/fuse and see if it fixes the problem?

**Comment by matpen** [comment]
@rhatdan I just tried with a fresh install of the latest [buildah master](https://github.com/containers/buildah/commit/20a33e0791eaeb4f13e23afba5cc73ad3e37bbe7), and can confirm that the following command completes successfully.

```
buildah bud --device /dev/fuse -t fuse-overlayfs -f ./Dockerfile.static .
```

Obviously, this requires the device `/dev/fuse` to be available on the host, which in turn requires to either install fuse or to run [this code](https://github.com/libfuse/libfuse/blob/f0e08cc700d629da2d46def8b620b0ed858cc0d9/util/install_helper.sh#L35) manually.

If so desired by the team, I can gladly provide a PR to add these steps to the [documentation](https://github.com/containers/fuse-overlayfs/blob/f8ba9ade20d107d3e6a3dc11b6e01116894d2387/README.md).

**Comment by rhatdan** [comment]
Sure we would love the PR.

**Comment by matpen** [comment]
I addressed the above in #121: please feel free to provide suggestions, and I will amend as necessary.
